test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/architecture/EmptyManagedKieServersWithSmartRouterAndControllerIntegrationTest.java

```diff
@@ -0,0 +1,291 @@
+    }
+
+    @Test
+    @Ignore("https://issues.jboss.org/browse/RHBPMS-4899 && https://issues.jboss.org/browse/JBPM-6623")
+    public void testConnectionBetweenDeployments() throws InterruptedException {
+        String kieServerIdABC = getKieServerId(kieServerClientABC);
+        String kieServerIdDEF = getKieServerId(kieServerClientDEF);
+        String kieServerIdGHI = getKieServerId(kieServerClientGHI);
+
+        logger.debug("Check architecture after start");
+        verifyServerTemplateContainsKieServers(kieControllerClient, SMART_ROUTER_ID, 1);
+        verifyServerTemplateContainsKieServers(kieControllerClient, kieServerIdABC, 1);
+        verifyServerTemplateContainsKieServers(kieControllerClient, kieServerIdDEF, 1);
+        verifyServerTemplateContainsKieServers(kieControllerClient, kieServerIdGHI, 1);
+
+        logger.debug("Deploy a container to kie server");
+        deployAndStartContainer(kieServerClientABC, CONTAINER_ID_ABC, deploymentScenario.getKieServerDeployments().get(0));
+        waitUntilKieServerLogsContain(deploymentScenario.getKieServerDeployments().get(0), "Container cont-id-abc (for release id org.kie.server.testing:definition-project-snapshot:1.0.0-SNAPSHOT) successfully started");
+        verifySmartRouterContainsKieServers(smartRouterAdminClient, 1, Arrays.asList(CONTAINER_ID_ABC), kieServerIdABC);
+        verifySmartRouterHostPerContainerContainsServers(smartRouterAdminClient, CONTAINER_ID_ABC, kieServerIdABC);
+        verifySmartRouterHostPerServerContainsContainers(smartRouterAdminClient, kieServerIdABC, CONTAINER_ID_ABC);
+
+        //check in smart router - kie servers and containers
+        verifySmartRouterContainsKieServers(smartRouterAdminClient, 1, Arrays.asList(CONTAINER_ID_ABC), kieServerIdABC);
+        verifySmartRouterHostPerContainerContainsServers(smartRouterAdminClient, CONTAINER_ID_ABC, kieServerIdABC);
+        verifySmartRouterHostPerServerContainsContainers(smartRouterAdminClient, kieServerIdABC, CONTAINER_ID_ABC);
+        logger.debug("Scale up all Kie server deployments to 3");
+        scaleKieServerTo(3, deploymentScenario.getKieServerDeployments());
+
+        logger.debug("Check all");
+        verifyServerTemplateContainsKieServers(kieControllerClient, SMART_ROUTER_ID, 1);
+        verifyServerTemplateContainsKieServers(kieControllerClient, kieServerIdABC, 3);
+        verifyServerTemplateContainsKieServers(kieControllerClient, kieServerIdDEF, 3);
+        verifyServerTemplateContainsKieServers(kieControllerClient, kieServerIdGHI, 3);
+        verifyServerTemplateContainsContainers(kieControllerClient, SMART_ROUTER_ID, CONTAINER_ID_ABC);
+        verifySmartRouterContainsKieServers(smartRouterAdminClient, 3, Arrays.asList(CONTAINER_ID_ABC), kieServerIdABC);
+    }
+
+    @Test
+    @Ignore("https://issues.jboss.org/browse/RHBPMS-4899 && https://issues.jboss.org/browse/JBPM-6623")
+    public void testWorkingWithSignalsAndTasksDeployments() {
```

Comment:
I think it would be better to separate signals and tasks to separate test methods. When they are combined together here then it makes the test less readable.

Reply:
If I separate test method, then test will take 5min longer. This time is consumed by OpenShift configuration, the most is taken by S2I builds.
I think it should be better to save some time here and separate all test to private methods and run only one test per class.